src/robot/libdocpkg/model.py

```diff
@@ -27,19 +28,25 @@ class LibraryDoc(object):
     def __init__(self, name='', doc='', version='', type='library',
                  scope='', named_args=True, doc_format=''):
         self.name = name
-        self.doc = doc
+        self.doc_format = doc_format
+        self._doc = doc
         self.version = version
         self.type = type
         self.scope = scope
         self.named_args = named_args
-        self.doc_format = doc_format
         self.inits = []
         self.keywords = []
 
     @setter
     def doc_format(self, format):
         return format or 'ROBOT'
 
+    @property
+    def doc(self):
+        if self.doc_format == 'ROBOT':
+            return add_toc(self._doc, self.inits)
```

Comment:
The ToC looked wrong if it would always create importing section, like [SL has](https://robotframework.org/SeleniumLibrary/SeleniumLibrary.html#Importing), so I made it configurable. But the `LibraryDoc` does not yet in this stage does not know, does library has a importing section or not. It is set later, like in here the [robotbuilder](https://github.com/robotframework/robotframework/blob/master/src/robot/libdocpkg/robotbuilder.py#L38). Therefore I did wonder, is my solution good one? Or should make bigger change and  change `LibraryDoc` __init__ to take in also the inits?

Reply:
Good question! My first reaction was that we could change `LibraryDocBuilder._get_doc` instead, but we then also needed to change other doc builders so probably changing model is better. 

What's the problem of using a property? If someone accesses `LibraryDoc.doc` before setting inits, then getting wrong TOC is kind of a user error. That's especially true if the property always recreates the TOC (i.e. doesn't set `sefl._doc`) and accessing `.doc` after inits have been set returns the correct TOC.

The above said, `LibraryDoc.__init__` getting new arguments `inits=None, keywords=None` would be fine. Even then someone could update `.inits` afterwards, but a property would always work correctly based on the current data.

Reply:
I already replied this elsewhere. I don't see a problem using a property.